src/tools/sssctl/sssctl_logs.c

```diff
@@ -604,7 +610,10 @@ errno_t sssctl_debug_level(struct sss_cmdline *cmdline,
 
 fini:
     talloc_free(ctx);
-    /* debug_as_string is not allocated but points to the command line. */
+    if (debug_as_string != NULL) {
+        free(debug_as_string);
```

Comment:
In other places you `free()` without check.

Reply:
Fixed